File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let arr = people.sort((a, b) => a - b);
    let cnt = sum = lt = 0;
    let rt = arr.length - 1;
    while (lt < rt) {
        sum = arr[lt] + arr[rt]
        if(sum <= limit) lt++;
        rt--;
        cnt++;
    }
    if (rt === lt) cnt++;
    return cnt;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    end = len(people) - 1
    people.sort()
    start = 0
    target = people[start]
    while end > start:
        sum = target + people[end]
        if sum <= limit:
            end -= 1
            start += 1
            target = people[start]
        else: 
            end -= 1
        answer += 1
        if end == start: answer += 1
    return answer